kong/tools/utils.lua

```diff
@@ -270,6 +270,28 @@ do
     end
   end
 
+
+  -- Recursively URL escape and format key and value
+  -- Handles nested arrays and tables
+  local function recursive_encode_args(parent_key, value, raw, no_array_indexes, query)
+    for sub_key, sub_value in pairs(value) do
```

Comment:
I think that we should duplicate the keys sorting logic at the beginning of `encode_args()` in this function as well. We want the result of the encoding to be identical in all platforms, which means that we cannot rely on `pairs`.